### ISSUE TITLE
firewall: need better feedback when adding and deleting services

### ISSUE THREAD

**Issue by andreasn** [cockpit-project/cockpit#9054]
During the [usability testing](https://github.com/cockpit-project/cockpit/wiki/Usability-test-Firewall) of the firewall, there was some uncertainty among some of the participants if a service had actually been added or deleted. So I think it needs a little more feedback somehow.

This came up in 50% of the tests (3 out of 6 participants)

**Comment by marusak** [comment]
I am afraid I don't follow what this issue is about. Can you please elaborate a bit more (if you still remember).
What feedback is needed?

**Comment by andreasn** [comment]
I don't quite remember, but let me see what I can dig up.

**Comment by andreasn** [comment]
**Here are the quotes from the study:**
**[p2]** _I feel like I didn't get enough confirmation._
**[p3]** _After I make a firewall rule change, I make sure. I want to see "on boot" and "active now". Is that change going to survive a reboot._

**And here is the note taker observation:**
_Even if adding a service port means activating the service it is not clear to the participants. Because the list says 'Allowed services' and the button on the right says 'Add services', but what users actually do using the 'Add services' dialog is adding 'ports'. So that's the mismatch that causes confusion. After adding the port, the participant sees it under the 'Allowed services' with no further feedback on if the service is active. Based on what they saw, the participants assumed that the port is now allowed and they went to 'Services' page to see if the service is active._

As far as I can see, we don't have full recordings of these sessions around any more, only highlights. Will dig deeper.

**Comment by marusak** [comment]
Right, so the thing is that users are not sure that when they add service, if it is already being used, or if they need to restart, or vice versa if it will survive reboot or if it is just temporary.
If I am not mistaken, it is both active straight away and will survive reboot. I don't think we have there anything that is just temporary or disabled. Thus I don't think it makes sense to add it to each and every item as that is just noise. Maybe we can just communicate it better in general that listed items are active - and that adding service will make it active an permanent.

We might rename `Allowed services` to `Active services`.
We might rename `service` to `firewall service` to distinguish it from `systemd service`.
We might add note into `Add service` dialog that it will `Enable and activate selected services`.

@Gundersanne can you please confirm or correct me if I said something incorrect above? Thanks!

**Comment by Gundersanne** [comment]
@marusak That's correct. When we're adding services it's added to both runtime and permanent configs.

That said if someone adds a runtime service, not through cockpit, and doesn't add that to the permanent config it'll disappear after reload/reboot.

There's also services that add themselves to the firewall at runtime, like libvirt, but that happens when libvirt itself starts so that will show up every time.

**Comment by marusak** [comment]
Thank you @Gundersanne 
So I think what we need is:

- rename `Allowed services` to `Active services`
- rename `service` to `firewall service` to distinguish it from `systemd service`
- add note into `Add service` dialog that it will `Enable and activate` selected services
- If some service is active but not enabled add badge to it saying it is not enabled

Does that make sense design?

**Comment by KKoukiou** [comment]
@marusak @Gundersanne I would like to add some more things I believe we should change here.

* The button 'Add services' needs also adjustment, since it allows the user to add add ports non associated with services as well. 
* Interfaces and addresses can also be added removed, and therefore I think they should be moved from the top of the card to a section where they have space to support these operations
*  Considering the two above, I would really create 4 Tabs, Services, Ports, Interfaces, Sources. The 'Add X' buttons needs to be part of the content of the tabs, and not the top of the card.
* For moving around services we can consider using https://www.patternfly.org/v4/components/dual-list-selector, I find it's a very good match here.

**Comment by KKoukiou** [comment]
> Thank you @Gundersanne
> So I think what we need is:
> 
>     * rename `Allowed services` to `Active services`

I am not super fond of this. Active services is also a systemd terminology and has nothing to do with firewall. firewall docs talk about, 'Adding a service to a zone' or 'Opening a service with firewalld'. So I would keep it to that direction, 'Added' makes more sense to me. 
> 
>     * rename `service` to `firewall service` to distinguish it from `systemd service`
> 
>     * add note into `Add service` dialog that it will `Enable and activate` selected services
> 
>     * If some service is active but not enabled add badge to it saying it is not enabled
> 
> 
> Does that make sense design wise?
